relayer/src/supervisor/scan.rs

```diff
@@ -339,6 +347,15 @@ impl<'a, Chain: ChainHandle> ChainScanner<'a, Chain> {
                     counterparty_connection_state,
                     client,
                 }) => {
+                    let counterparty_chain_id = client.client_state.chain_id();
+                    init_telemetry(
+                        &chain.id(),
+                        &client.client_id,
+                        &counterparty_chain_id,
+                        channel_id,
+                        port_id,
+                    );
+
```

Comment:
As the `client.client_state.chain_id()` clones the `ChainId` value I was wondering if:

1. It is more optimal to directly pass the reference `&client.client_state.chain_id()` to the `init_telemetry` call:
```
 init_telemetry(
    ...
    &client.client_state.chain_id()
    ...
    )
```
2. It is more optimal to declare a variable as it is done here
3. It does not matter as the compiler will handle both cases the same way

@romac @soareschen Do you have an idea ?

Reply:
I think using a variable improves readability. 

Regarding reference, I used a reference so `init_telemetry()` would not consume `counterparty_chain_id` and eventually reuse it (line [684](https://github.com/informalsystems/ibc-rs/blob/25bf5d7f0836c05da22d2bcfdb0a40f85f71cc0a/relayer/src/supervisor/scan.rs#L684) and [689](https://github.com/informalsystems/ibc-rs/blob/25bf5d7f0836c05da22d2bcfdb0a40f85f71cc0a/relayer/src/supervisor/scan.rs#L689)).